sql/size_utils.sql

```diff
@@ -421,6 +421,49 @@ BEGIN
 END;
 $BODY$;
 
+-- Convenience function to list all hypertables and their space usage
+--
+-- Returns:
+-- schema_name      - Schema name of the hypertable
+-- table_name       - Table name of the hypertable
+-- num_dimensions   - Number of dimensions
+-- num_chunks       - Number of chunks
+-- table_size       - pretty output of table_size
+-- index_size       - pretty output of index_size
+-- toast_size       - pretty output of toast_size
+-- total_size       - pretty output of total_bytes
+
+CREATE OR REPLACE FUNCTION show_hypertables()
```

Comment:
FWIW, I think this should, like most of our functions now, be in C. This is where we are moving towards and we should not regress.

This would also overlap with functionality already implemented by @Ngalstyan4 in hos `show_chunks` PR. There he has functions to get a list of hypertables

Reply:
Can you post a link to his function to list hypertables cause i couldnt find it.

Reply:
I think having this as a view is better then a function also makes the filtering discussion moot. Regarding permissions i think it should only list those the user has access to similar to other views in information_schema, so the default output should be all hypertables the user can access. It won't fail if the user cant access the tables but it wont show space usage for those.